### PR TITLE
Unsafe syntax

### PR DESCRIPTION
**Pull request by metromoxie**
@devd This implements the policy syntax. The descriptions of the policy options are meant as drafts to describe what they do, but they'll need to be integrated into individual sections later.

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -420,13 +425,22 @@ for the name and value of the header are described by the following ABNF
 grammar [[!RFC5234]]:
 
 <pre dfn-type="grammar" link-type="grammar">
-    suborigin = 1*( <a>LOWERALPHA</a> / <a>DIGIT</a> / "-" )
+    <dfn>suborigin-name</dfn> = 1*( <a>LOWERALPHA</a> / <a>DIGIT</a> / "-" )
+    <dfn>suborigin-policy-option</dfn> = "'unsafe-postmessage-send'"
+                              / "'unsafe-postmessage-receive'"
```
Comment (devd) [inline index.bs]: I thought we were making this also a single unsafe-postmessage in v1?

File: index.bs [hunk 2]
```diff
@@ -616,10 +630,40 @@ algorithm:
...
+NOT use these options unless they are required to make their application work.
+
+The values of <a link-type="grammar">suborigin-policy-option</a> that may be
+present in a <a>suborigin policy</a> have the following effects:
+
+* '<dfn>`unsafe-postmessage-send`</dfn>' When a message is sent <i>from</i> a suborigin
+  namespace with <a>`unsafe-postmessage-send`</a> set, the `event.origin` value
+  of the receiver should be set to the serialized physical origin, not the
+  serialized suborigin value. However, the `event.suborigin` field should still
+  be set to the name of the suborigin namespace.
```
Comment (devd) [inline index.bs]: A note on how people can use this if they want would be great

File: index.bs [hunk 3]
```diff
@@ -616,10 +630,40 @@ algorithm:
...
+  serialized suborigin, if the frame has an execution context with a suborigin
+  where the scheme, host, and port match the `target`, but it also has a
+  suborigin namespace, if <a>`unsafe-postmessage-receive`</a> is set, it will
+  still receive the message.
+
+* '<dfn>`unsafe-cookies`</dfn>' When an execution context with a suborigin
+  namespace has <a>`unsafe-cookies`</a> set, the
+  execution context should not have a fresh cookie jar for the suborigin
+  namespace, and instead, the cookie jar should be shared with the null
+  suborigin for the execution context.
```
Comment (devd) [inline index.bs]: It would be nice to explicitly flesh out the set-cookie vs document.cookie use cases. Maybe leave a TODO?

**Comment by devd** [comment]
mostly lgtm; minor nits/qs

File: index.bs (hunk 1 above)
Comment (metromoxie) [inline index.bs]: Er, I guess I thought the opposite :-) I don't care too much, although they seem like different enough to me that they'd warrant separate directives (principle of least privilege yadda yadda).

File: index.bs (hunk 2 above)
Comment (metromoxie) [inline index.bs]: Go idea. Will add shortly.

File: index.bs (hunk 3 above)
Comment (metromoxie) [inline index.bs]: Will add shortly.

**Comment by metromoxie** [comment]
@devd, check out the updates. If you think it still looks good, feel free to go ahead and merge. I'm happy to keep discussing `unsafe-postmessage` vs `unsafe-postmessage-send` + `unsafe-postmessage-receive`, but maybe we can make that ta separate issue?

File: index.bs (hunk 1 above)
Comment (devd) [inline index.bs]: well, in general we can add it in later anyhow. My concern is that i am not sure in how many cases one is fine and not the other. If it is, that probably means someone somewhere is not checking the origin or not setting the origin correctly.

File: index.bs [hunk 4]
```diff
@@ -616,10 +630,77 @@ algorithm:
...
+  Continuing the case in the above example, `https://example.com/maps` is a
+  mapping application that is commonly embedded in other sites. It provides a
+  `postMessage()` based API to place locations of the embedder's choosing on the
+  map. `httsp://example.com` would like to run the application in a suborigin
+  named "maps".
+
+  In response to queries to the API, `https://example.com/maps` may send
+  messages back to the embedder if, for example, a user clicks on one of the
+  locations. However, since the embedder may be legacy and not be aware of
+  suborigins, when it checks the sender of the message, if it seem
```
Comment (devd) [inline index.bs]: s/seem/sees/

when it checks the event.origin property of the postMessage seems more concerete than "checks sender of the message"

File: index.bs [hunk 5]
```diff
@@ -616,10 +630,77 @@ algorithm:
...
+
+* '<dfn>`unsafe-cookies`</dfn>' When an execution context with a suborigin
+  namespace has <a>`unsafe-cookies`</a> set, the
+  execution context should not have a fresh cookie jar for the suborigin
+  namespace, and instead, the cookie jar should be shared with the null
+  suborigin for the execution context.
+
+  Issue: TODO: Flesh out document.cookie vs set cookie use cases. Also, write an
+  example that makes clear that this is extremely dangerous and should not be
+  used if you have Real Deal auth cookies used.
```
Comment (devd) [inline index.bs]: real deal auth or csrf cookies

File: index.bs [hunk 6]
```diff
@@ -616,10 +630,77 @@ algorithm:
...
+
+Issue: TODO: These opt-out descriptions should probably be moved to the
```
Comment (devd) [inline index.bs]: I actually like them all in a separate section.

**Comment by devd** [comment]
lgtm; fix the minor typo and merge it.

File: index.bs (hunk 1 above)
Comment (metromoxie) [inline index.bs]: Fair point. If everyone that wants to use one wants the other as well, might as well make it one directive. Not sure how to best figure that out though.

Comment (metromoxie) [inline index.bs]: Yeah, this is kind of an "unsafe-embedded-frame-api" directives really.

File: index.bs (hunk 4 above)
Comment (metromoxie) [inline index.bs]: Sounds good, although I'm changing it to "...the event.origin property of the `MessageEvent`..."

File: index.bs (hunk 5 above)
Comment (metromoxie) [inline index.bs]: Done.

**Comment by metromoxie** [comment]
Last push squashed the commits.

**Pull request merged**